src/kvstore/raftex/Host.cpp

```diff
@@ -106,37 +106,22 @@ folly::Future<cpp2::AppendLogResponse> Host::appendLogs(
         std::lock_guard<std::mutex> g(lock_);
 
         auto res = checkStatus();
-
-        if (logId == logIdToSend_) {
-            // This is a re-send or a heartbeat. If there is an
-            // ongoing request, we will just return SUCCEEDED
-            if (requestOnGoing_) {
-                LOG(INFO) << idStr_ << "Another request is onging,"
-                                       "ignore the re-send request";
-                cpp2::AppendLogResponse r;
-                r.set_error_code(cpp2::ErrorCode::SUCCEEDED);
-                return r;
-            }
-        } else {
-            // Otherwise, logId has to be greater
-            if (logId < logIdToSend_) {
-                LOG(INFO) << idStr_ << "The log has been sended";
-                cpp2::AppendLogResponse r;
-                r.set_error_code(cpp2::ErrorCode::SUCCEEDED);
-                return r;
-            }
+        if (logId <= lastLogIdSent_) {
```

Comment:
a niubility change.
but I have a question, which case will cause logId <= lastLogIdSent_? and if appear the case logId == lastLogIdSent_, we return success directly, and this will cause collectNSucceeded() return success directly in "RaftPart::replicateLogs", and the current thread will commit the log,   but in the other thread do really thing that replica log to other part by network and but failed, then the log just exist in leader?

Reply:
> which case will cause logId <= lastLogIdSent_?

Currently,  quorum policy means there are some hosts requests maybe not in the order what we want.  Especially, for learner,  we can't ensure the order.  

> but in the other thread do really thing that replica log to other part by network and but failed, then the log just exist in leader?

Not exactly,  lastLogIdSent_ is used to record the last log id the leader send to some follower.  So if logId <= lastLogIdSent_,  it means the log has been received by the follower.

Reply:
But in the code and raft logic, the logid is Monotonically increasing, and the lastLogIdSent_ is always set by the prevLogId which is less than logId, even if the host is a learner. so is it some thing wrong when in case " logId <= lastLogIdSent_"

Reply:
> lastLogIdSent_ is always set by the prevLogId which is less than logId

Not exactly,  the lastLogIdSent_ only be updated when received response from follower.  So the logId leader want to send maybe less than lastLogIdSent_

Reply:
I see it update immediate in Host::appendLogs before call appendLogsInternal.

Reply:
The logic has been updated in this pr